pkg/cli/adbtunnel.go

```diff
@@ -233,24 +233,31 @@ func adbTunnelAgentLoop(flags *ADBTunnelFlags, c *command, args []string, opts *
 		return err
 	}
 
-	devProps := deviceProperties{
-		serviceURL: flags.ServiceURL,
-		zone:       flags.Zone,
-		host:       flags.host,
-		device:     device,
+	devSpec := DeviceSpec{
+		ServiceURL: flags.ServiceURL,
+		Zone:       flags.Zone,
+		Host:       flags.host,
+		Device:     device,
 	}
-	logger, err := createLogger(opts.InitialConfig.ADBControlDirExpanded(), devProps)
-	if err != nil {
+
+	controlDir := opts.InitialConfig.ADBControlDirExpanded()
+	port, controller, err := findOrCreateTunnel(controlDir, devSpec, service)
+	if port < 0 {
```

Comment:
can we handle this using type assertion instead, i.e  if err.(FindOrCreateTunnelFatalError)?

Reply:
separated non-fatal from fatal errors instead.